cli/src/commands/auth.rs

```diff
@@ -130,12 +130,14 @@ pub async fn handle_auth(
             },
             Err(error) => Err(error).context("User registration failed"),
         },
-        Some(("login", _)) => match handle_auth_login(config, config_path).await {
-            Ok(_) => {
-                print_user_success!("{}", "User login successful");
-                Ok(ExitCode::Ok.into())
-            },
-            Err(error) => Err(error).context("User login failed"),
+        Some(("login", matches)) => {
+            match handle_auth_login(config, config_path, matches.get_flag("reauth")).await {
+                Ok(_) => {
+                    print_user_success!("{}", "User login successful");
+                    Ok(ExitCode::Ok.into())
+                },
+                Err(error) => Err(error).context("User login failed"),
+            }
         },
```

Comment:
I'm not sure I really care in this example, but by convention I feel like we tend to pass through the commands more often than not instead of getting the flag directly on the toplevel.

Since these toplevel structures tend to grow kinda unwieldy, it's usually nice to have all the stuff specific about each endpoint defined inside of it. It also means that future things pulled out of the CLI arguments don't require a change here anymore.

Reply:
I agree with this, but if anything, I'd say it would be a good idea to do a refactoring pass and adapt all the other command handlers to have a similar behavior where command matches don't get passed thorugh. Not a huge ticket nor super urgent, but I'd suggest we think about it as a tech debt repayment at some point in the future.

Reply:
Yeah I'm sure other areas are problematic too, but considering we should already be pretty good in most areas I'd refrain from introducing further tech debt if we can avoid it.